### PR TITLE
Sports: Fix missing possession in NFL

### PR DESCRIPTION
**Pull request by mrshu**
###### What does your Pull Request do (check all that apply)?

Choose the most relevant items and use the following title template to name
your Pull Request.

- [ ] New Instant Answer: **`New {IA Name} Instant Answer`**
- [x] Improvement
    - [x] Bug fix: **`Sports: Fix missing possession in NFL`**
    - [ ] Enhancement: **`{IA Name}: {Description of Improvements}`**
- [ ] Non–Instant Answer
    - [ ] Other (Role, Template, Test, Documentation, etc.): **`{SpiceRole/Templates/Tests/Docs}: {Short Description}`**

###### Description of changes

* The previous implementation assumed that the possession data will be
  always available. This might not be the case, especially when a game
  gets canceled or delayed.

  This commit fixes the situation by skipping the part of the code that
  requires the possession to be filled in.

###### Which issues (if any) does this fix?

See above

###### People to notify (@mention interested parties)

@b1ake 

---

Instant Answer Page: https://duck.co/ia/view/sports

[Maintainer](http://docs.duckduckhack.com/maintaining/guidelines.html): @b1ake 

* The previous implementation assumed that the possession data will be
  always available. This might not be the case, especially when a game
  gets canceled or delayed.

  This commit fixes the situation by skipping the part of the code that
  requires the possession to be filled in.

Signed-off-by: mr.Shu <mr@shu.io>

### CODE REVIEW

**Comment by daxtheduck** [comment]
Metadata placeholder post
            
*If you haven't already, please edit your first comment to include a link to the IA page at the bottom.*
            
---
            
**How to find your IA page**
- New Instant Answer
  -- follow the steps [here](https://duck.co/ia/new_ia) to create your IA page.
              
- Existing Instant Answer
  -- find your IA page from this [list](https://duck.co/ia)

File: share/spice/sports/nfl/games/sports_nfl_games.js [hunk 1]
```diff
@@ -65,6 +65,11 @@
                     // fill boxscore with placeholder data
                     attrs.score = games.fillBoxscore(attrs.score, ops);
 
+                    // finish here if there is no possession data
+                    if (!attrs.score.possession) {
```
Comment (b1ake) [inline share/spice/sports/nfl/games/sports_nfl_games.js]: Nice! Is it possible that `attr.scores` is even undefined? If so, maybe we should catch that here too.

Comment (mrshu) [inline share/spice/sports/nfl/games/sports_nfl_games.js]: @b1ake given the definition of [`fillBoxscore`](https://github.com/duckduckgo/zeroclickinfo-spice/blob/04aaf11a04664c390accb8d5676618eacfa4b950/share/spice/sports/sports.js#L207) it seems that it easily might be.

Will update the PR, thanks for catching it!

**Comment by b1ake** [comment]
👍

**Pull request merged**